cpp/bench/ann/src/cuvs/cuvs_mg_cagra_wrapper.h

```diff
@@ -87,7 +86,7 @@ class cuvs_mg_cagra : public algo<T>, public algo_gpu {
   std::unique_ptr<algo<T>> copy() override;
 
  private:
-  raft::device_resources handle_;
+  raft::device_resources_snmg clique_;
```

Comment:
`device_resources_snmg` is a type of `raft::resource`. The pattern here is to accept the `raft::resources` instance and call the free function inside `raft::resource::nccl_resource` underneath to get the clique out of the resource. This way the APIs don't need to look any different.

Reply:
We should be able to store `raft::resources` here now, right? The idea being that `has_resource_factory` for the `NCCL_CLIQUE` will be `true` when it's an SNMG type? (else, we'll need to add some other type of state, along with an accessor in the `resources/nccl_clique.hpp` for `is_multi_gpu()` and `is_multi_node()`

Reply:
Maybe it's not a huge deal since this file is specific to mg. Could we at least rename the file to `snmg`, though? (Just in case we end up adding an mnmg then we won't have a clash).